src/properties/forest.c

```diff
@@ -0,0 +1,215 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
```

Comment:
I think the copyright headers should have like 3 extra newlines in them, which should be the case in most other files.